client/json-rpc/src/client.rs

```diff
@@ -149,15 +149,9 @@ impl JsonRpcAsyncClientError {
     pub fn is_retriable(&self) -> bool {
         match self {
             JsonRpcAsyncClientError::ClientError(e) => {
-                // TODO: Add following condition when is_request is added to reqwest stable release
-                // // Error sending request
-                //  if e.is_request() {
-                //      return true;
-                //  }
-                if e.is_timeout() {
+                if e.is_timeout() || e.is_request() {
```

Comment:
what is this is_request?

also, please add test

Reply:
is_request() is set to true whenever there is a problem sending the request. According to the reqwest docs `/// Returns true if the error is related to the request`

I don't think its possible to unit test this since the constructor for this error type is private to the crate.
I locally tested this by prematurely disconnecting the upstream server and verified that `e.is_request()` is set to true.